fix(components): [popper] fix arrow overflow issue

xiaowhang opened the pull request:
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

Fixes #17451

## Issue

When the reference element is significantly larger than the popper element and the placement is set to directional variants (xxx-start 或 xxx-end), the arrow can overflow beyond the edges of the popper.

## Solution

Added popperArrowProp to popperContentProps to ensure usePopper correctly applies arrow options according to the [Popper.js documentation on arrow padding](https://popper.js.org/docs/v2/modifiers/arrow/#padding).

## Before

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiBjbGFzcz1cInRvb2x0aXAtYmFzZS1ib3hcIj5cbiAgICA8ZWwtdG9vbHRpcCBlZmZlY3Q9XCJkYXJrXCIgY29udGVudD1cIkJvdHRvbSBMZWZ0IHByb21wdHMgSW5mb1wiIHBsYWNlbWVudD1cImJvdHRvbS1zdGFydFwiPlxuICAgICAgPGVsLWlucHV0IC8+XG4gICAgPC9lbC10b29sdGlwPlxuICAgIDxlbC10b29sdGlwIGVmZmVjdD1cImRhcmtcIiBjb250ZW50PVwiQm90dG9tIFJpZ2h0IHByb21wdHMgSW5mb1wiIHBsYWNlbWVudD1cImJvdHRvbS1lbmRcIj5cbiAgICAgIDxlbC1pbnB1dCAvPlxuICAgIDwvZWwtdG9vbHRpcD5cbiAgICA8ZWwtdG9vbHRpcCBlZmZlY3Q9XCJkYXJrXCIgY29udGVudD1cIlRvcCBMZWZ0IHByb21wdHMgSW5mb1wiIHBsYWNlbWVudD1cInRvcC1zdGFydFwiPlxuICAgICAgPGVsLWlucHV0IC8+XG4gICAgPC9lbC10b29sdGlwPlxuICAgIDxlbC10b29sdGlwIGVmZmVjdD1cImRhcmtcIiBjb250ZW50PVwiVG9wIFJpZ2h0IHByb21wdHMgSW5mb1wiIHBsYWNlbWVudD1cInRvcC1lbmRcIj5cbiAgICAgIDxlbC1pbnB1dCAvPlxuICAgIDwvZWwtdG9vbHRpcD5cbiAgICA8ZWwtdG9vbHRpcCBlZmZlY3Q9XCJkYXJrXCIgY29udGVudD1cIlJpZ2h0IExlZnQgcHJvbXB0cyBJbmZvXCIgcGxhY2VtZW50PVwicmlnaHQtc3RhcnRcIj5cbiAgICAgIDxlbC1pbnB1dCBzdHlsZT1cImhlaWdodDogMTIwcHg7XCIgLz5cbiAgICA8L2VsLXRvb2x0aXA+XG4gICAgPGVsLXRvb2x0aXAgZWZmZWN0PVwiZGFya1wiIGNvbnRlbnQ9XCJSaWdodCBSaWdodCBwcm9tcHRzIEluZm9cIiBwbGFjZW1lbnQ9XCJyaWdodC1lbmRcIj5cbiAgICAgIDxlbC1pbnB1dCBzdHlsZT1cImhlaWdodDogMTIwcHg7XCIgLz5cbiAgICA8L2VsLXRvb2x0aXA+XG4gICAgPGVsLXRvb2x0aXAgZWZmZWN0PVwiZGFya1wiIGNvbnRlbnQ9XCJSaWdodCBSaWdodCBwcm9tcHRzIEluZm9cIiBwbGFjZW1lbnQ9XCJyaWdodC1lbmRcIj5cbiAgICAgIDxlbC1zZWxlY3Qgc2l6ZT1cImxhcmdlXCIgLz5cbiAgICA8L2VsLXRvb2x0aXA+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cblxuPHN0eWxlPlxuLnRvb2x0aXAtYmFzZS1ib3gge1xuICB3aWR0aDogNjAwcHg7XG59XG48L3N0eWxlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

## After

![image](https://github.com/user-attachments/assets/404cb1e6-c650-4383-9cad-2ed126cd83d6)

xiaowhang committed:
fix(components): [popper] fix arrow overflow issue

Add popperArrowProp to popperContentProps to ensure usePopper
correctly applies arrow options, preventing the arrow from
overflowing the popper container edges.

xiaowhang commented:
/label Components:[popper]

xiaowhang commented:
/label Components:Popper

xiaowhang commented:
> 🧪 Playground Preview: https://element-plus.run/?pr=20049 Please comment the example via this playground if needed.

Like this: [Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IEVsZW1lbnRQbHVzIH0gZnJvbSAnQGVsZW1lbnQtcGx1cy9pY29ucy12dWUnXG5pbXBvcnQgeyB2ZXJzaW9uIGFzIGVwVmVyc2lvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBtc2cgPSByZWYoJ0hlbGxvIFdvcmxkIScpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwidG9vbHRpcC1iYXNlLWJveFwiPlxuICAgIDxlbC10b29sdGlwIGVmZmVjdD1cImRhcmtcIiBjb250ZW50PVwiQm90dG9tIExlZnQgcHJvbXB0cyBJbmZvXCIgcGxhY2VtZW50PVwiYm90dG9tLXN0YXJ0XCI+XG4gICAgICA8ZWwtaW5wdXQgLz5cbiAgICA8L2VsLXRvb2x0aXA+XG4gICAgPGVsLXRvb2x0aXAgZWZmZWN0PVwiZGFya1wiIGNvbnRlbnQ9XCJCb3R0b20gUmlnaHQgcHJvbXB0cyBJbmZvXCIgcGxhY2VtZW50PVwiYm90dG9tLWVuZFwiPlxuICAgICAgPGVsLWlucHV0IC8+XG4gICAgPC9lbC10b29sdGlwPlxuICAgIDxlbC10b29sdGlwIGVmZmVjdD1cImRhcmtcIiBjb250ZW50PVwiVG9wIExlZnQgcHJvbXB0cyBJbmZvXCIgcGxhY2VtZW50PVwidG9wLXN0YXJ0XCI+XG4gICAgICA8ZWwtaW5wdXQgLz5cbiAgICA8L2VsLXRvb2x0aXA+XG4gICAgPGVsLXRvb2x0aXAgZWZmZWN0PVwiZGFya1wiIGNvbnRlbnQ9XCJUb3AgUmlnaHQgcHJvbXB0cyBJbmZvXCIgcGxhY2VtZW50PVwidG9wLWVuZFwiPlxuICAgICAgPGVsLWlucHV0IC8+XG4gICAgPC9lbC10b29sdGlwPlxuICAgIDxlbC10b29sdGlwIGVmZmVjdD1cImRhcmtcIiBjb250ZW50PVwiUmlnaHQgTGVmdCBwcm9tcHRzIEluZm9cIiBwbGFjZW1lbnQ9XCJyaWdodC1zdGFydFwiPlxuICAgICAgPGVsLWlucHV0IHN0eWxlPVwiaGVpZ2h0OiAxMjBweDtcIiAvPlxuICAgIDwvZWwtdG9vbHRpcD5cbiAgICA8ZWwtdG9vbHRpcCBlZmZlY3Q9XCJkYXJrXCIgY29udGVudD1cIlJpZ2h0IFJpZ2h0IHByb21wdHMgSW5mb1wiIHBsYWNlbWVudD1cInJpZ2h0LWVuZFwiPlxuICAgICAgPGVsLWlucHV0IHN0eWxlPVwiaGVpZ2h0OiAxMjBweDtcIiAvPlxuICAgIDwvZWwtdG9vbHRpcD5cbiAgICA8ZWwtdG9vbHRpcCBlZmZlY3Q9XCJkYXJrXCIgY29udGVudD1cIlJpZ2h0IFJpZ2h0IHByb21wdHMgSW5mb1wiIHBsYWNlbWVudD1cInJpZ2h0LWVuZFwiPlxuICAgICAgPGVsLXNlbGVjdCBzaXplPVwibGFyZ2VcIiAvPlxuICAgIDwvZWwtdG9vbHRpcD5cblxuICAgIDxlbC1kcm9wZG93biBwbGFjZW1lbnQ9XCJib3R0b20tZW5kXCI+XG4gICAgICA8ZWwtYnV0dG9uPiBUaGlzIGlzIGEgbG9vb29vb29vb29vb29vZyB0ZXh0IDwvZWwtYnV0dG9uPlxuICAgICAgPHRlbXBsYXRlICNkcm9wZG93bj5cbiAgICAgICAgPGVsLWRyb3Bkb3duLW1lbnU+XG4gICAgICAgICAgPGVsLWRyb3Bkb3duLWl0ZW0+VGhlIEFjdGlvbiAxc3Q8L2VsLWRyb3Bkb3duLWl0ZW0+XG4gICAgICAgICAgPGVsLWRyb3Bkb3duLWl0ZW0+VGhlIEFjdGlvbiAyc3Q8L2VsLWRyb3Bkb3duLWl0ZW0+XG4gICAgICAgICAgPGVsLWRyb3Bkb3duLWl0ZW0+VGhlIEFjdGlvbiAzc3Q8L2VsLWRyb3Bkb3duLWl0ZW0+XG4gICAgICAgIDwvZWwtZHJvcGRvd24tbWVudT5cbiAgICAgIDwvdGVtcGxhdGU+XG4gICAgPC9lbC1kcm9wZG93bj5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c3R5bGU+XG4udG9vbHRpcC1iYXNlLWJveCB7XG4gIHdpZHRoOiA2MDBweDtcbn1cbjwvc3R5bGU+XG4iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL3ByZXZpZXctMjAwNDktZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9kaXN0L2luZGV4LmNzcycsICdodHRwczovL3ByZXZpZXctMjAwNDktZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS90aGVtZS1jaGFsay9kYXJrL2Nzcy12YXJzLmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsU2V0dGxlZChzdHlsZXMpXG59XG4iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMjAwNDktZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9kaXN0L2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctMjAwNDktZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9kaXN0L2luZGV4LmNzcyJ9fQ==)

xiaowhang commented:
事实上popper相关的组件应该都能正常使用，如：[Playground Link](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IHN0eWxlPVwid2lkdGg6IDYwMHB4OyBtYXJnaW4tYm90dG9tOiA4MHB4O1wiPlxuICAgIDxlbC10b29sdGlwIGVmZmVjdD1cImRhcmtcIiBjb250ZW50PVwiQm90dG9tIFJpZ2h0IHByb21wdHMgSW5mb1wiIHBsYWNlbWVudD1cImJvdHRvbS1lbmRcIj5cbiAgICAgIDxlbC1pbnB1dCBwbGFjZWhvbGRlcj1cIkRlZmF1bHQgY29uZmlndXJhdGlvblwiIC8+XG4gICAgPC9lbC10b29sdGlwPlxuICA8L2Rpdj5cblxuICA8ZGl2PlxuICAgIDxlbC1kcm9wZG93biBwbGFjZW1lbnQ9XCJib3R0b20tZW5kXCIgc3R5bGU9XCJtYXJnaW4tYm90dG9tOiA2MHB4O1wiPlxuICAgICAgPGVsLWJ1dHRvbj4gVGhpcyBpcyBhIGxvb29vb29vb29vb29vb29vb29nIGRyb3Bkb3duIHRleHQgPC9lbC1idXR0b24+XG4gICAgICA8dGVtcGxhdGUgI2Ryb3Bkb3duPlxuICAgICAgICA8ZWwtZHJvcGRvd24tbWVudT5cbiAgICAgICAgICA8ZWwtZHJvcGRvd24taXRlbT5kcm9wZG93bi1pdGVtPC9lbC1kcm9wZG93bi1pdGVtPlxuICAgICAgICA8L2VsLWRyb3Bkb3duLW1lbnU+XG4gICAgICA8L3RlbXBsYXRlPlxuICAgIDwvZWwtZHJvcGRvd24+XG4gIDwvZGl2PlxuXG4gIDxkaXY+XG4gICAgPGVsLXBvcGNvbmZpcm0gdGl0bGU9XCJBcmUgeW91IHN1cmUgdG8gZGVsZXRlIHRoaXM/XCIgcGxhY2VtZW50PVwiYm90dG9tLWVuZFwiPlxuICAgICAgPHRlbXBsYXRlICNyZWZlcmVuY2U+XG4gICAgICAgIDxlbC1idXR0b24+IFRoaXMgaXMgYSBsb29vb29vb29vb29vb29vb29vZyBwb3Bjb25maXJtIHRleHQgPC9lbC1idXR0b24+XG4gICAgICA8L3RlbXBsYXRlPlxuICAgIDwvZWwtcG9wY29uZmlybT5cbiAgPC9kaXY+XG5cbiAgPGRpdiBzdHlsZT1cIndpZHRoOiA2MDBweDsgbWFyZ2luOiA4MHB4IDA7XCI+XG4gICAgPGVsLXRvb2x0aXAgZWZmZWN0PVwiZGFya1wiIGNvbnRlbnQ9XCJCb3R0b20gUmlnaHQgcHJvbXB0cyBJbmZvXCIgcGxhY2VtZW50PVwiYm90dG9tLWVuZFwiIDpwb3BwZXItb3B0aW9ucz1cIntcbiAgICAgIG1vZGlmaWVyczogW1xuICAgICAgICB7XG4gICAgICAgICAgbmFtZTogJ2Fycm93JyxcbiAgICAgICAgICBvcHRpb25zOiB7XG4gICAgICAgICAgICBwYWRkaW5nOiAyMCxcbiAgICAgICAgICB9LFxuICAgICAgICB9LFxuICAgICAgXSxcbiAgICB9XCI+XG4gICAgICA8ZWwtaW5wdXQgcGxhY2Vob2xkZXI9XCJPdmVycmlkZSB0aGUgY29uZmlndXJhdGlvbiB1c2luZyBwb3BwZXItb3B0aW9ucy5cIiAvPlxuICAgIDwvZWwtdG9vbHRpcD5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuIiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIGNvbnN0IHN0eWxlcyA9IFsnaHR0cHM6Ly9wcmV2aWV3LTIwMDQ5LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvZGlzdC9pbmRleC5jc3MnLCAnaHR0cHM6Ly9wcmV2aWV3LTIwMDQ5LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvdGhlbWUtY2hhbGsvZGFyay9jc3MtdmFycy5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbFNldHRsZWQoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTIwMDQ5LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvZGlzdC9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTIwMDQ5LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvZGlzdC9pbmRleC5jc3MifX0=)
## Tooltip：
Before:
![image](https://github.com/user-attachments/assets/7ba270b3-15fc-4215-99f8-e6155b7cb53c)
After:
![image](https://github.com/user-attachments/assets/0577b6f5-79e6-401f-926e-a4e5d917a26d)


## Popover：
Before:
![image](https://github.com/user-attachments/assets/ebd95e2a-b64f-4a7d-94e6-44730905924b)
After:
![image](https://github.com/user-attachments/assets/b28cbe8a-8545-420d-8142-7c13c50282b4)


## Popconfirm：
Before:
![image](https://github.com/user-attachments/assets/8eaf8430-fb28-415d-8734-5636bd9e5f71)
After:
![image](https://github.com/user-attachments/assets/f1d03a5e-a3fc-4da1-93d8-2fee602d38bf)

## use popper-options：
popper-options也能正确覆盖配置，这里设置padding为20（默认为5）
![image](https://github.com/user-attachments/assets/405f515b-a964-4399-a70d-5821ff9fec83)

问题的原因应该是在之前popper的集成时并没有使用popperArrowProps的参数进行createPopper，将其引入popperContentProps确保createPopper时能正确使用arrow的默认配置

warmthsea reviewed (commented):
It seems that the idea is correct 👍 

But there are several problems:

1. The document should add `arrowOffset`. In fact, there is this feature, but it is not recorded in the document.

2. There seems to be a problem with the code after the change, [code 1](https://github.com/xiaowhang/element-plus/blob/dev/packages/components/popper/src/arrow.vue#L24-L34), [code 2](https://github.com/xiaowhang/element-plus/blob/dev/packages/components/popper/src/constants.ts#L21-L25)

![image](https://github.com/user-attachments/assets/f61c3cd9-1aee-492d-9f7f-a61d9efa6010)

xiaowhang committed:
fix(components): [popper] change arrowOffset to ref for reactivity

xiaowhang committed:
feat(components): [tooltip] add arrow-offset property

- Allow users to control the padding of the tooltip arrow
- Prevents the arrow from touching the popper's edge

Reference: https://popper.js.org/docs/v2/modifiers/arrow/#padding

btea commented on packages/components/popper/src/content.vue:
```diff
@@ -84,7 +84,7 @@ const {
 
 const formItemContext = inject(formItemContextKey, undefined)
-const arrowOffset = ref<number>()
+const arrowOffset = ref<number>(props.arrowOffset)
```
Here we may add a watch `props.arrowOffset` to match the dynamic changes of `props.arrowOffset`.

xiaowhang committed:
fix(components): [popper] add reactive watch for arrowOffset prop

xiaowhang commented on packages/components/popper/src/content.vue:
```diff
@@ -84,7 +84,15 @@ const {
+  () => props.arrowOffset,
+  (val) => {
+    arrowOffset.value = val
+    updatePopper(false)
```
我发现这里的 `arrowOffset` 似乎是一个无效的属性，它通过 `POPPER_CONTENT_INJECTION_KEY` 提供给packages/components/popper/src/arrow.vue 但是并没有被使用。我本来是想删除它的，但是这会导致 arrow.test.tsx 的 `'should update the offset after props changed'`无法通过。

这是我的第一次 PR，我不确定更改原有的测试用例是否是一个正确的实现，如果有不妥之处，欢迎指正，非常感谢你的耐心和指导！

btea replied:
看起来确实如此，把那个测试用例删掉吧。

xiaowhang committed:
feat(components): [popper] remove unused arrowOffset prop

Removes unused arrowOffset prop and related logic
- Removes arrowOffset from PopperContentInjectionContext
- Removes arrowOffset prop watching and state management
- Cleans up related test cases

btea commented on docs/en-US/component/tooltip.md:
```diff
@@ -173,6 +173,7 @@ tooltip/append-to
 | offset                    | offset of the Tooltip                                                                                                                                   | ^[number]                                                                                                                                                                   | 12                |
 | transition                | animation name                                                                                                                                          | ^[string]                                                                                                                                                                   | —                 |
 | popper-options            | [popper.js](https://popper.js.org/docs/v2/) parameters                                                                                                  | ^[object]refer to [popper.js](https://popper.js.org/docs/v2/) doc                                                                                                           | {}                |
+| arrow-offset ^(2.9.7)     | Controls the offset (padding) of the tooltip’s arrow relative to the popper.                                                                            | ^[number]                                                                                                                                                                   | 5                 |
```
```suggestion
| arrow-offset ^(2.9.8)     | Controls the offset (padding) of the tooltip’s arrow relative to the popper.                                                                            | ^[number]                                                                                                                                                                   | 5                 |
```

xiaowhang committed:
docs(components): [tooltip] update version requirement

update version requirement for arrow-offset prop

xiaowhang committed:
Merge branch 'dev' into dev

xiaowhang committed:
docs(components): [tooltip] update `arrow-offset` version

xiaowhang commented:
@warmthsea Could you please help merge this PR when you have time? Thanks a lot!

kooriookami commented on docs/en-US/component/tooltip.md:
```diff
@@ -173,6 +173,7 @@ tooltip/append-to
 | offset                    | offset of the Tooltip                                                                                                                                   | ^[number]                                                                                                                                                                   | 12                |
 | transition                | animation name                                                                                                                                          | ^[string]                                                                                                                                                                   | —                 |
 | popper-options            | [popper.js](https://popper.js.org/docs/v2/) parameters                                                                                                  | ^[object]refer to [popper.js](https://popper.js.org/docs/v2/) doc                                                                                                           | {}                |
+| arrow-offset ^(2.9.9)     | Controls the offset (padding) of the tooltip’s arrow relative to the popper.                                                                            | ^[number]                                                                                                                                                                   | 5                 |
```
```suggestion
| arrow-offset ^(2.9.10)     | Controls the offset (padding) of the tooltip’s arrow relative to the popper.                                                                            | ^[number]                                                                                                                                                                   | 5                 |
```

warmthsea committed:
Update docs/en-US/component/tooltip.md

Co-authored-by: kooriookami <38392315+kooriookami@users.noreply.github.com>

warmthsea reviewed (approved):
LGTM, Thanks!
